src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h

```diff
@@ -71,6 +71,26 @@ void set_wrap_mode(
     mode[1] = set_wrap_mode(wrap_tmode);
 }
 
+void set_wrap_mode(
+    int wrap_mode[3],
+    output string mode[3])
+{
+    mode[0] = set_wrap_mode(wrap_mode[0]);
```

Comment:
Something is wrong here. `set_wrap_mode()` doesn't return anything.

Reply:
There is a string return type set above. We could bypass this completely if we had the ability to associate a string value with another string (or any other value really) into the OSL token for the *mapper* options, but although there is apparently nothing forbidding it in the spec, it's not working in appleseed-maya so far.

Reply:
I see. I would suggest changing the name of the first `set_wrap_mode()` function (the one that returns a string): it doesn't set anything, instead it retrieves the string version of a wrap mode. Maybe `get_wrap_mode_name()`?

Reply:
Good point, done.